katana: add page (#12619)
tldr-pages/tldr: pages/common/katana.md

# katana

> > A next-generation crawling and spidering tool.

Reviewer: ```suggestion
> A fast crawler focused on execution in automation pipelines offering both headless and non-headless crawling.
> See also: `gau`, `scrapy`, `waymore`.
```
- "web crawling" is a synonymous for "web spidering", right? I took this description from the usage section of their README
- I added a "See also" section for some similar tools

Author: My description was taken from the About section (top right) on the project's GitHub page. I am ok with your suggestion.

> More information: <https://github.com/projectdiscovery/katana>.

> - Crawl a [u]RL or a list of URLs:

Reviewer: ```suggestion
- Crawl a list of URLs:
```

`katana -list {{https://example.com,https://google.com,...}}`

> - Crawl using headless mode using Chromium:

Reviewer: ```suggestion
- Crawl a [u]RL using headless mode using Chromium:
```

> - Crawl using [h]ead[l]ess mode using Chromium:

Reviewer 2: ```suggestion
- Crawl using headless mode using Chromium:
```

IMO it is non-standard that short options can be more than one character, so I'd just leave it out.

Author: @gutjuri This practice of using multi chars for short options exists in other places, e.g. `nuclei`, `httpx`

Reviewer: It's a different case there, it's because they are for abbreviated options. We only use this when the long options is abbreviated or when there is a long option with these letters.

Author: I am finding it hard to follow. I think we should clarify the style guide. Anyway, I will accept the suggestion by gutjuri.

`katana -u {{https://example.com}} -headless`

> - Use [p]a[s]sive sources (waybackarchive, commoncrawl, and alienvault) for URL discovery:

Reviewer 3: ```suggestion
- Use [p]a[s]sive sources (Wayback Machine, Common Crawl, and alienvault) for URL discovery:
```
Shouldn't this be the Wayback machine or is it something else?

> - Use [p]a[s]sive sources (waybackarchive, commoncrawl, and alienvault) for URL discovery:

Reviewer 2: ```suggestion
- Use passive sources (waybackarchive, commoncrawl, and alienvault) for URL discovery:
```

`cat {{example.com}} | katana -passive`

- Pass requests through a proxy (http/socks5) and use custom [H]eaders from a file:

`katana -proxy {{http://127.0.0.1:8080}} -headers {{path/to/headers.txt}} -u {{https://example.com}}`

> - Specify crawling [s]trategy, [d]epth of subdirectories to crawl, and rate limiting (requests per second):

Reviewer 2: ```suggestion
- Specify the crawling [s]trategy, [d]epth of subdirectories to crawl, and rate limiting (requests per second):
```
